### DISCUSSION TITLE
Upper bound limit on CP-SAT: MODEL_INVALID

### DISCUSSION THREAD

**Discussion by acheong08** [google/or-tools#4353] [CP-SAT questions] [answered]
**What version of OR-Tools and what language are you using?**
Version: 9.10.4067
Language: Python

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**

CP-SAT

**What operating system (Linux, Windows, ...) and version?**

Linux (Fedora 40)

**What did you do?**
Set upper bound of `IntVar` to `100_000_000_000_000`

**What did you expect to see**

Tell me the upper bound is too high and the maximum possible upper bound

**What did you see instead?**

<details>

```
Model Invalid
Possible integer overflow in constraint: linear {
  ...
}

CpSolverResponse summary:
status: MODEL_INVALID
objective: 0
best_bound: 0
integers: 0
booleans: 0
conflicts: 0
branches: 0
propagations: 0
integer_propagations: 0
restarts: 0
lp_iterations: 0
walltime: 0.0181558
usertime: 0.0181558
deterministic_time: 0
gap_integral: 0

```

</details>

**Comment by lperron** [comment, marked as answer]
This is the expected behavior.

The activity of a linear constraint cannot overflow an int64_t.
